src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/MsBuild/MsBuildUtilitiesTest.cs

```diff
@@ -0,0 +1,362 @@
+
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.VisualStudio.ProjectSystem
+{
+    [ProjectSystemTrait]
+    public class MsBuildUtilitiesTest
+    {
+        [Fact]
+        public void MsBuildUtilities_GetProperty_MissingProperty()
+        {
+            string projectXml =
+@"<Project>
+  <PropertyGroup>
+     <MyProperty>MyPropertyValue</MyProperty>
+  </PropertyGroup>
+</Project>";
+
+            using (var project = new MsBuildTempProjectFile(projectXml))
+            {
+                var property = MsBuildUtilities.GetProperty(project.Project, "NonExistantProperty");
+                Assert.Null(property);
+            }
+        }
+
+        [Fact]
+        public void MsBuildUtilities_GetProperty_ExistentProperty()
+        {
+            string projectXml =
+@"<Project>
+  <PropertyGroup>
+     <MyProperty>MyPropertyValue</MyProperty>
+  </PropertyGroup>
+</Project>";
+
+            using (var project = new MsBuildTempProjectFile(projectXml))
+            {
+                var property = MsBuildUtilities.GetProperty(project.Project, "MyProperty");
+                Assert.NotNull(property);
+            }
+        }
+
+        [Fact]
+        public void MsBuildUtilities_GetPropertyValues_SingleValue()
+        {
+            var values = MsBuildUtilities.GetPropertyValues("MyPropertyValue");
+            Assert.Equal(1, values.Length);
+            Assert.Equal("MyPropertyValue", values[0]);
+        }
+
+        [Fact]
+        public void MsBuildUtilities_GetPropertyValues_MultipleValues()
+        {
+            var values = MsBuildUtilities.GetPropertyValues("1;2");
+            Assert.Equal(2, values.Length);
+            Assert.Equal("1", values[0]);
+            Assert.Equal("2", values[1]);
+        }
+
+        [Fact]
+        public void MsBuildUtilities_GetPropertyValues_NonDefaultDelimiter()
+        {
+            var values = MsBuildUtilities.GetPropertyValues("1|2", '|');
+            Assert.Equal(2, values.Length);
+            Assert.Equal("1", values[0]);
+            Assert.Equal("2", values[1]);
+        }
+
+        [Fact]
+        public void MsBuildUtilities_GetPropertyValues_EmptyValues()
+        {
+            var values = MsBuildUtilities.GetPropertyValues("1;   ;;;2");
+            Assert.Equal(2, values.Length);
+            Assert.Equal("1", values[0]);
+            Assert.Equal("2", values[1]);
+        }
+
+        [Fact]
+        public void MsBuildUtilities_GetOrAddProperty_NoGroups()
+        {
+            using (var project = new MsBuildTempProjectFile())
+            {
+                MsBuildUtilities.GetOrAddProperty(project.Project, "MyProperty");
+                Assert.Equal(1, project.Project.Properties.Count);
+                Assert.Equal(1, project.Project.PropertyGroups.Count);
+                var group = project.Project.PropertyGroups.First();
```

Comment:
Nit: Line breaks after a bunch of asserts tells the user the rest of the test is testing a different component. Improves readability.